### PR TITLE
Add flag to check if rules have been provided by config and check it …

### PR DESCRIPTION
**Pull request by CarlesDD**
…to enable ASM_DATA

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/test/appsec/remote_config/index.spec.js
Comment (simon-id) [inline packages/dd-trace/test/appsec/remote_config/index.spec.js:1]: can we add a test for the regression ? ie: "should enable when using default rules" or something like that

File: packages/dd-trace/src/config.js [hunk 1]
```diff
@@ -480,6 +480,7 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
     this.appsec = {
       enabled: DD_APPSEC_ENABLED,
       rules: DD_APPSEC_RULES ? safeJsonParse(maybeFile(DD_APPSEC_RULES)) : require('./appsec/recommended.json'),
+      customRulesProvided: DD_APPSEC_RULES !== undefined,
```
Comment (simon-id) [inline packages/dd-trace/src/config.js:483]: It could also be null tho

**Review by simon-id — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (CarlesDD) [reply packages/dd-trace/src/config.js:483]: I cannot see how. 

Value is coalesced [here](https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/config.js#L302) and the output will be a string or undefined, but maybe I'm missing something?

File: packages/dd-trace/test/appsec/remote_config/index.spec.js [hunk 2]
```diff
@@ -136,15 +136,26 @@ describe('Remote Config enable', () => {
...
+        config.appsec = { enabled: true, customRulesProvided: true }
         remoteConfig.enable(config)
         remoteConfig.enableAsmData(config.appsec)
 
         expect(rc.updateCapabilities).to.not.have.been.called
         expect(rc.on).to.not.have.been.calledWith('ASM_DATA')
       })
 
+      it('should enable when using default rules', () => {
+        config.appsec = { enabled: true, customRulesProvided: false }
```
Comment (simon-id) [inline packages/dd-trace/test/appsec/remote_config/index.spec.js:149]:
```suggestion
        config.appsec = { enabled: true, rules: {}, customRulesProvided: false }
```

File: packages/dd-trace/src/config.js (hunk 1 above)
Comment (simon-id) [reply packages/dd-trace/src/config.js:483]: What I'm saying is this line should be consistent with line 482 above, both should make the same check, otherwise it's inconsistent and may lead to stupid edge cases

File: packages/dd-trace/test/appsec/remote_config/index.spec.js [hunk 3]
```diff
@@ -136,15 +136,26 @@ describe('Remote Config enable', () => {
         expect(rc.on).to.not.have.been.calledWith('ASM_DATA')
       })
 
-      it('should not activate if rules is configured', () => {
-        config.appsec = { enabled: true, rules: './path/rules.json' }
+      it('should not enable when custom appsec rules are provided', () => {
+        config.appsec = { enabled: true, customRulesProvided: true }
```
Comment (simon-id) [inline packages/dd-trace/test/appsec/remote_config/index.spec.js:140]:
```suggestion
        config.appsec = { enabled: true, rules: {}, customRulesProvided: true }
```

File: packages/dd-trace/test/config.spec.js
Comment (simon-id) [inline packages/dd-trace/test/config.spec.js:1]: I feel like the tests `should left undefined appsec rules if user rules file could not be loaded` and `should skip appsec config files if they do not exist` do the same thing no ? I think we should merge them ?

**Review by simon-id — APPROVED** [review]

**Review by tlhunter — APPROVED** [review]

**Review by Qard — APPROVED** [review]
